### DISCUSSION TITLE
How to get everything working after installation

### DISCUSSION THREAD

**Discussion by JRSofty** [abraunegg/onedrive#1987] [Q&A] [answered]
Ok so I followed the instructions for installation on a Ubuntu 22.04 https://github.com/abraunegg/onedrive/blob/master/docs/ubuntu-package-install.md#distribution-ubuntu-2204

The first thing I noticed was that the `onedrive.service` file listed in the Known Issues section doesn't exist after the installation. https://github.com/abraunegg/onedrive/blob/master/docs/ubuntu-package-install.md#1-the-onedrive-client-will-automatically-startup-post-authentication-without-any-further-actions

As I didn't see any sort of indication that the application is started I started the application in my terminal and I'm provided a link for authorization, ok so I open the link, login to Microsoft and double check on my other computer that the Onedrive Client for Linux is actually allowed to access my onedrive. I then copy the resulting URL from my browser, and pasted it back into the terminal and received some obsure error that it wasn't properly authorized (yet I see it as authorized in my Microsoft account). 

SO I figured I have to reboot my computer for it to take affect, and do so. Once I've logged back into my computer, I don't see that the application is running, and so once again call it via terminal and once again it says I need to authenticate. 

So what am I doing wrong and how do I get this onedrive client working on my Ubuntu 22.04 system.

**Comment by abraunegg** [comment]
@JRSofty 
> The first thing I noticed was that the onedrive.service file listed in the Known Issues section doesn't exist after the installation. https://github.com/abraunegg/onedrive/blob/master/docs/ubuntu-package-install.md#1-the-onedrive-client-will-automatically-startup-post-authentication-without-any-further-actions

Correct .. as this was *fixed* in the v2.4.18 release and the documentation needs to be updated - as a number of folk are still running older clients where this is still valid, but only valid until they upgrade to v2.4.18.

> As I didn't see any sort of indication that the application is started I started the application in my terminal and I'm provided a link for authorization, ok so I open the link, login to Microsoft and double check on my other computer that the Onedrive Client for Linux is actually allowed to access my onedrive. I then copy the resulting URL from my browser, and pasted it back into the terminal and received some obsure error that it wasn't properly authorized (yet I see it as authorized in my Microsoft account).
> ...
> So what am I doing wrong and how do I get this onedrive client working on my Ubuntu 22.04 system.

How did you authenticate the client - what was the *exact* command you ran?

To have the application *run* automatically, you need to configure the systemd services - refer to https://github.com/abraunegg/onedrive/blob/master/docs/USAGE.md#running-onedrive-as-a-system-service

**Comment by JRSofty** [comment]
Hi thanks for the quick reply.

From my terminal I called `onedrive` and the application gave me a link to open. I copied the link into my browser and it brought me to the Microsoft login.

**Reply by abraunegg** [reply]
OK .. and you said there was a 'weird' response .. what was this *exact* response ?

**Comment by abraunegg** [comment, marked as answer]
@JRSofty 
So you have a *clear* output of the various steps you need to take:

**Install:**
As per https://github.com/abraunegg/onedrive/blob/master/docs/ubuntu-package-install.md#distribution-ubuntu-2204
```
Welcome to Ubuntu 22.04 LTS (GNU/Linux 5.15.0-35-generic x86_64)

 * Documentation:  https://help.ubuntu.com
 * Management:     https://landscape.canonical.com
 * Support:        https://ubuntu.com/advantage

0 updates can be applied immediately.

Last login: Sat Jun  4 07:55:37 2022 from X.X.X.X
alex@ubuntu-22-04:~$ wget -qO - https://download.opensuse.org/repositories/home:/npreining:/debian-ubuntu-onedrive/xUbuntu_22.04/Release.key | gpg --dearmor | sudo tee /usr/share/keyrings/obs-onedrive.gpg > /dev/null
[sudo] password for alex: 
alex@ubuntu-22-04:~$ echo "deb [arch=$(dpkg --print-architecture) signed-by=/usr/share/keyrings/obs-onedrive.gpg] https://download.opensuse.org/repositories/home:/npreining:/debian-ubuntu-onedrive/xUbuntu_22.04/ ./" | sudo tee /etc/apt/sources.list.d/onedrive.list
deb [arch=amd64 signed-by=/usr/share/keyrings/obs-onedrive.gpg] https://download.opensuse.org/repositories/home:/npreining:/debian-ubuntu-onedrive/xUbuntu_22.04/ ./
alex@ubuntu-22-04:~$ sudo apt-get update
Hit:1 http://au.archive.ubuntu.com/ubuntu jammy InRelease
Hit:2 http://au.archive.ubuntu.com/ubuntu jammy-updates InRelease
Hit:3 http://au.archive.ubuntu.com/ubuntu jammy-backports InRelease
Hit:4 http://security.ubuntu.com/ubuntu jammy-security InRelease               
Get:5 https://download.opensuse.org/repositories/home:/npreining:/debian-ubuntu-onedrive/xUbuntu_22.04 ./ InRelease [1,604 B]
Get:6 https://download.opensuse.org/repositories/home:/npreining:/debian-ubuntu-onedrive/xUbuntu_22.04 ./ Packages [812 B]
Fetched 2,416 B in 3s (809 B/s)
Reading package lists... Done
alex@ubuntu-22-04:~$ sudo apt install onedrive
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
The following additional packages will be installed:
  libllvm11 libphobos2-ldc-shared98
The following NEW packages will be installed:
  libllvm11 libphobos2-ldc-shared98 onedrive
0 to upgrade, 3 to newly install, 0 to remove and 0 not to upgrade.
Need to get 21.8 MB of archives.
After this operation, 92.7 MB of additional disk space will be used.
Do you want to continue? [Y/n] y
Get:1 http://au.archive.ubuntu.com/ubuntu jammy/universe amd64 libllvm11 amd64 1:11.1.0-6 [19.6 MB]
Get:3 http://au.archive.ubuntu.com/ubuntu jammy/universe amd64 libphobos2-ldc-shared98 amd64 1:1.28.0-1ubuntu1 [1,487 kB]
Get:2 https://download.opensuse.org/repositories/home:/npreining:/debian-ubuntu-onedrive/xUbuntu_22.04 ./ onedrive 2.4.18-1~np1 [720 kB]
Fetched 21.8 MB in 4s (4,917 kB/s)                       
Selecting previously unselected package libllvm11:amd64.
(Reading database ... 176661 files and directories currently installed.)
Preparing to unpack .../libllvm11_1%3a11.1.0-6_amd64.deb ...
Unpacking libllvm11:amd64 (1:11.1.0-6) ...
Selecting previously unselected package libphobos2-ldc-shared98:amd64.
Preparing to unpack .../libphobos2-ldc-shared98_1%3a1.28.0-1ubuntu1_amd64.deb ...
Unpacking libphobos2-ldc-shared98:amd64 (1:1.28.0-1ubuntu1) ...
Selecting previously unselected package onedrive.
Preparing to unpack .../onedrive_2.4.18-1~np1_amd64.deb ...
Unpacking onedrive (2.4.18-1~np1) ...
Setting up libllvm11:amd64 (1:11.1.0-6) ...
Setting up libphobos2-ldc-shared98:amd64 (1:1.28.0-1ubuntu1) ...
Setting up onedrive (2.4.18-1~np1) ...
Processing triggers for man-db (2.10.2-1) ...
Processing triggers for libc-bin (2.35-0ubuntu3) ...
alex@ubuntu-22-04:~$ 
```

**First Run and Authentication:**
As per https://github.com/abraunegg/onedrive/blob/master/docs/USAGE.md#authorize-the-application-with-your-onedrive-account
```
Welcome to Ubuntu 22.04 LTS (GNU/Linux 5.15.0-35-generic x86_64)

 * Documentation:  https://help.ubuntu.com
 * Management:     https://landscape.canonical.com
 * Support:        https://ubuntu.com/advantage

0 updates can be applied immediately.

Last login: Sat Jun  4 08:09:40 2022 from X.X.X.X
alex@ubuntu-22-04:~$ onedrive
Configuring Global Azure AD Endpoints
Authorize this app visiting:

https://login.microsoftonline.com/common/oauth2/v2.0/authorize?client_id=d50ca740-c83f-4d1b-b616-12c519384f0c&scope=Files.ReadWrite%20Files.ReadWrite.all%20Sites.Read.All%20Sites.ReadWrite.All%20offline_access&response_type=code&prompt=login&redirect_uri=https://login.microsoftonline.com/common/oauth2/nativeclient

Enter the response uri: https://login.microsoftonline.com/common/oauth2/nativeclient?code=M.R3_BAY.this_is_redacted

Application has been successfully authorised, however no additional command switches were provided.

Please use 'onedrive --help' for further assistance in regards to running this application.

alex@ubuntu-22-04:~$ 
```

**Enable systemd service:**
As per https://github.com/abraunegg/onedrive/blob/master/docs/USAGE.md#onedrive-service-running-as-a-non-root-user-via-systemd-with-notifications-enabled-arch-ubuntu-debian-opensuse-fedora
```
Welcome to Ubuntu 22.04 LTS (GNU/Linux 5.15.0-35-generic x86_64)

 * Documentation:  https://help.ubuntu.com
 * Management:     https://landscape.canonical.com
 * Support:        https://ubuntu.com/advantage

0 updates can be applied immediately.

Last login: Sat Jun  4 08:12:13 2022 from X.X.X.X
alex@ubuntu-22-04:~$ systemctl --user enable onedrive
Created symlink /home/alex/.config/systemd/user/default.target.wants/onedrive.service → /usr/lib/systemd/user/onedrive.service.
alex@ubuntu-22-04:~$ systemctl --user start onedrive
alex@ubuntu-22-04:~$ journalctl --user-unit=onedrive -f
Jun 04 08:14:25 ubuntu-22-04 onedrive[2937]: Creating local directory: random_files/KLAZRQ9v2eVD0zjPxgqpoxRrtrMIIiMP
Jun 04 08:14:25 ubuntu-22-04 onedrive[2937]: Creating local directory: random_files/phr7Pm3hr3E29Kx3EPUUcMJfrxBVfP83
Jun 04 08:14:25 ubuntu-22-04 onedrive[2937]: Creating local directory: random_files/1QqpVWGEGo9gk8XIbjsrKO5wwRWKXeDF
Jun 04 08:14:25 ubuntu-22-04 onedrive[2937]: Creating local directory: random_files/0KsMcVf0VtxLAnNHiFrjYtYZ5gaMJExg
Jun 04 08:14:25 ubuntu-22-04 onedrive[2937]: Creating local directory: random_videos/HkfC8b1nQqpTX5qnxoLBrUMRZvLZJjNP
Jun 04 08:14:25 ubuntu-22-04 onedrive[2937]: Creating local directory: random_videos/nWTLfs9seZygNp8OjVcb8AUQeMw3JJAs
Jun 04 08:14:25 ubuntu-22-04 onedrive[2937]: Creating local directory: random_videos/AYu1wuseF06B1FLVaS9dth6aWnD2WkO1
Jun 04 08:14:25 ubuntu-22-04 onedrive[2937]: Creating local directory: random_videos/w7btxzBZKQjX3djI2KRAh6YYkujiN0GM
Jun 04 08:14:25 ubuntu-22-04 onedrive[2937]: Creating local directory: random_videos/7hX6WsMmpPbJBQvanYqFruwpclMZFfml
Jun 04 08:14:28 ubuntu-22-04 onedrive[2937]: Downloading file random_videos/LV8pkgZlO5OlraSdSSEaHJmPmwjBqGLg/test_video_0.mpg ... done.
Jun 04 08:14:30 ubuntu-22-04 onedrive[2937]: Downloading file random_videos/LV8pkgZlO5OlraSdSSEaHJmPmwjBqGLg/test_video_2.mpg ... done.
Jun 04 08:14:32 ubuntu-22-04 onedrive[2937]: Downloading file random_videos/LV8pkgZlO5OlraSdSSEaHJmPmwjBqGLg/test_video_3.mpg ... done.
Jun 04 08:14:32 ubuntu-22-04 onedrive[2937]: Downloading file random_videos/XHNAFGv5Ki7afJ90flJHVE8eeSCxzcHy/test_video_0.mpg ...
^C
alex@ubuntu-22-04:~$ 
alex@ubuntu-22-04:~$ systemctl --user status onedrive  
● onedrive.service - OneDrive Free Client
     Loaded: loaded (/usr/lib/systemd/user/onedrive.service; enabled; vendor preset: enabled)
     Active: active (running) since Sat 2022-06-04 08:14:18 AEST; 31s ago
       Docs: https://github.com/abraunegg/onedrive
   Main PID: 2937 (onedrive)
      Tasks: 3 (limit: 4630)
     Memory: 42.2M
        CPU: 2.646s
     CGroup: /user.slice/user-1000.slice/user@1000.service/app.slice/onedrive.service
             └─2937 /usr/bin/onedrive --monitor

Jun 04 08:14:32 ubuntu-22-04 onedrive[2937]: Downloading file random_videos/LV8pkgZlO5OlraSdSSEaHJmPmwjBqGLg/test_video_3.mpg ... done.
Jun 04 08:14:32 ubuntu-22-04 onedrive[2937]: Downloading file random_videos/XHNAFGv5Ki7afJ90flJHVE8eeSCxzcHy/test_video_0.mpg ...
Jun 04 08:14:35 ubuntu-22-04 onedrive[2937]: [3.2K blob data]
Jun 04 08:14:35 ubuntu-22-04 onedrive[2937]: done.
Jun 04 08:14:37 ubuntu-22-04 onedrive[2937]: Downloading file random_videos/XHNAFGv5Ki7afJ90flJHVE8eeSCxzcHy/test_video_2.mpg ... done.
Jun 04 08:14:40 ubuntu-22-04 onedrive[2937]: Downloading file random_videos/LV8pkgZlO5OlraSdSSEaHJmPmwjBqGLg/test_video_4.mpg ... done.
Jun 04 08:14:43 ubuntu-22-04 onedrive[2937]: Downloading file random_videos/119kgj48pQVDAj6EloBkLZNtKsx8jmox/test_video_0.mpg ... done.
Jun 04 08:14:44 ubuntu-22-04 onedrive[2937]: Downloading file random_videos/LV8pkgZlO5OlraSdSSEaHJmPmwjBqGLg/test_video_1.mpg ... done.
Jun 04 08:14:48 ubuntu-22-04 onedrive[2937]: Downloading file random_videos/XHNAFGv5Ki7afJ90flJHVE8eeSCxzcHy/test_video_4.mpg ... done.
Jun 04 08:14:49 ubuntu-22-04 onedrive[2937]: Downloading file random_videos/119kgj48pQVDAj6EloBkLZNtKsx8jmox/test_video_3.mpg ... done.
alex@ubuntu-22-04:~$ 

```

**Reply by JRSofty** [reply]
Hi sorry I didn't get back to you last night. 

The error I'm getting is an Internet connectivity issue. Which doesn't make sense seeing I'm on the internet. 

Everything from your examples above looks fine until the authentication.

I'm going to uninstall and start over from the beginning and see if I can get it up and running. 

![Screenshot from 2022-06-04 07-55-13](https://user-images.githubusercontent.com/1228089/171986624-6eb13d04-95b3-4b11-96f6-f0d29c9b2956.png)

So I `sudo apt remove onedrive` then `sudo apt-get update` followed by a `sudo apt install onedrive` and it all installed nicely it seems and when I ran the `onedrive` it says it can't connect (see screen shot). So I performed a `onedrive --logout` to remove the old authentication information. Then ran `onedrive` again and I get the url. Which I opened, it opened in my browser and I went through my authentication and the page redirected to an empty page which I took the URL from and put it in and now it's saying it cannot connect to the Onedrive service.

![Screenshot from 2022-06-04 08-04-44](https://user-images.githubusercontent.com/1228089/171987221-dbacd226-7d1a-4dfd-abd3-ab7fcdb4738c.png)

It ran with this message for nearly 5 minutes and finally at the end it said 
![Screenshot from 2022-06-04 08-19-02](https://user-images.githubusercontent.com/1228089/171987270-1f74e6a6-b5b8-4e17-9728-41b3da303d9b.png)

Maybe it's my device that just doesn't like the setup. I'm completely at a loss for the errors I'm getting. No big worries, it'll have to wait until I return from vacation in a couple of weeks. Then I can further attempt to troubleshoot the problem.

**Reply by abraunegg** [reply]
@JRSofty 
Please read https://github.com/abraunegg/onedrive/wiki/Frequently-Asked-Questions#cannot-connect-to-microsoft-onedrive-service---network-connection-issue in relation to this issue

**Reply by JRSofty** [reply]
Nope that doesn't seem to be the problem result returned is just fine:
![Screenshot from 2022-06-04 08-34-44](https://user-images.githubusercontent.com/1228089/171987757-b7b3d3c2-15dd-44b5-919f-60a0c694a0a8.png)

**Reply by abraunegg** [reply]
That error is *specifically* dealing with Internet access and resolution. This is 100% accurate and correct.

You have a DNS (bad or slow response)  or Firewall rule problem.

Given that the curl test now works - does the authorisation now work ?

**Reply by JRSofty** [reply]
That's got it. Thanks for your help.
